crates/containerd-shim-wasm/src/sandbox/containerd.rs

```diff
@@ -90,47 +98,51 @@ impl Client {
                 .map_err(|err| ShimError::Others(err.to_string()))?
                 .into_inner()
                 .container
-                .ok_or(ShimError::Others(format!("failed to get image for container {}", container_name.to_string())))?
+                .ok_or(ShimError::Others(format!(
+                    "failed to get image for container {}",
+                    container_name.to_string()
+                )))?
                 .image;
             Ok(image)
         })
     }
 
-    // load module will query the containerd store to find an image that has an ArtifactType of WASM OCI Artifact
+    // load module will query the containerd store to find an image that has an OS of type 'wasm'
     // If found it continues to parse the manifest and return the layers that contains the WASM modules
-    // and possibly other configuration artifacts
-    pub fn load_modules(&self, containerd_id: impl ToString) -> Result<Vec<oci::OciArtifact>> {
+    // and possibly other configuration layers.
+    pub fn load_modules(&self, containerd_id: impl ToString) -> Result<Vec<oci::WasmLayer>> {
         let image_name = self.get_image(containerd_id.to_string())?;
         let digest = self.get_image_content_sha(image_name)?;
         let manifest = self.read_content(digest)?;
         let manifest = manifest.as_slice();
         let manifest = ImageManifest::from_reader(manifest)?;
 
-        let artifact_type = manifest
-            .artifact_type()
+        let os = manifest
+            .config()
+            .platform()
             .as_ref()
-            .ok_or(ShimError::Others("manifest is not an OCI Artifact".to_string()))?;
+            .ok_or(ShimError::Others("failed to extract OS".to_string()))?
+            .os();
 
-        match artifact_type {
-            MediaType::Other(s) if s == COMPONENT_ARTIFACT_TYPE || s == MODULE_ARTIFACT_TYPE => {
-                log::info!("manifest with OCI Artifact of type {s}");
+        match os {
+            Os::Other(s) if s == "wasm" => {
```

Comment:
Isn't the os `wasi` / `wasip1`, while `wasm` is the arch?

Also, some images use `wasm32`, but that's ok because this would only apply to new images.

Reply:
yup! fixed.